allennlp/data/data_loaders/multitask_data_loader.py

```diff
@@ -0,0 +1,264 @@
+    desired, to have the sampling and/or scheduling behavior be dependent on the current state of
+    training.
+
+    While it is not necessarily required, this `DatasetReader` was designed to be used alongisde a
+    `MultiTaskModel`, which can handle instances coming from different datasets.  If your datasets
+    are similar enough (say, they are all reading comprehension datasets with the same format), or
+    your model is flexible enough, then you could feasibly use this `DataLoader` with a normal,
+    non-multitask `Model`.
+
+    Registered as a `DataLoader` with name "multitask".
+
+    # Parameters
+
+    reader: `MultiTaskDatasetReader`
+    data_path: `Dict[str, str]`
+        One file per underlying dataset reader in the `MultiTaskDatasetReader`, which will be passed
+        to those readers to construct one `DataLoader` per dataset.
+    batch_size: `int`
+        The number of instances (from any dataset) that should be combined together into a single
+        batch.  See also the `batch_size_multiplier` argument for additional control over exactly
+        how batch size is computed.
+    scheduler: `MultiTaskScheduler`, optional (default = `RoundRobinScheduler`)
+        The `scheduler` determines how instances are ordered within an epoch.  By default, we'll
+        select one instance from each dataset in turn, trying to ensure as uniform a mix of datasets
+        as possible.
+    sampler: `MultiTaskEpochSampler`, optional (default = `None`)
+        Only used if `instances_per_epoch` is not `None`. If we need to select a subset of the data
+        for an epoch, this `sampler` will tell us with what proportion we should sample from each
+        dataset.  For instance, we might want to focus more on datasets that are underperforming in
+        some way, by having those datasets contribute more instances this epoch than other datasets.
+    batch_size_multiplier: `Dict[str, int]`, optional (default = `None`)
+        If this is not `None`, it specifies how much of the batch an instance from each dataset
+        takes up.  That is, if this is 1 for every dataset (which is the default), then batch size
+        is computed as normal.  If dataset "A" has a value of 2 in this dictionary, than each
+        instance from dataset "A" counts as 2 instances for the purposes of computing batch size.
+        This option is available to you to account for the fact that some operations might be *much*
+        less costly than others (e.g., if you are multitasking a coref model with a simple document
+        classification model).  If you want finer-grained control than integer multiples, just scale
+        all values in the dictionary and the batch size accordingly (e.g., to achieve a 1:1.5 ratio,
+        double your batch size and use 2 and 3 as your multipliers).  If you use this, you're on
+        your own as far as figuring out how it interacts with optimization behavior.
```

Comment:
What is this for?

If we need it, couldn't we make this easier to specify by making it a `Dict[str, float]`, and just saying we'll do it by the fraction of the sum of the given floats?

Reply:
> What is this for?

I gave an example in the docstring, so I'm not sure what you're asking here.

> If we need it, couldn't we make this easier to specify by making it a `Dict[str, float]`, and just saying we'll do it by the fraction of the sum of the given floats?

Hmm, yeah, I was originally thinking that we'd have to have the batch size always be an integer, and so we'd need this to stay as integers.  But looking through the logic, it looks like having float batch sizes works alright for the internal logic here, and the floats would never get seen outside of this class, so we should be ok.  I'll change this.